File: use-cases/10-data-model/expected/render/go/domain.go
package domain

type Customer struct {
	Country string `json:"country"`
	Currency string `json:"currency"`
	ID string `json:"id"`
	LedgerID int64 `json:"ledgerId"`
	Name string `json:"name"`
}

type Invoice struct {
	Currency string `json:"currency"`
	GrossCents int64 `json:"grossCents"`
	ID string `json:"id"`
	NetCents int64 `json:"netCents"`
	OrderID string `json:"orderId"`
	TaxCents int64 `json:"taxCents"`
}

type Order struct {
	CustomerID string `json:"customerId"`
	ID string `json:"id"`
	Lines []OrderLine `json:"lines"`
	Placed *string `json:"placed,omitempty"`
	Status *string `json:"status,omitempty"`
}

type OrderLine struct {
	AmountCents int64 `json:"amountCents"`
	Qty int64 `json:"qty"`
	Sku string `json:"sku"`
	UnitCents int64 `json:"unitCents"`
}

File: use-cases/10-data-model/expected/render/domain.ts
export interface Customer {
  country: string;
  currency: string;
  id: string;
  ledgerId: number;
  name: string;
}

export interface Invoice {
  currency: string;
  grossCents: number;
  id: string;
  netCents: number;
  orderId: string;
  taxCents: number;
}

export interface Order {
  customerId: string;
  id: string;
  lines: unknown[];
  placed: string;
  status: string;
}

export interface OrderLine {
  amountCents: number;
  qty: number;
  sku: string;
  unitCents: number;
}
